### PR TITLE
Structure_Engine: Adding method for checking objects on loads id assignment

### LINKED ISSUE #875: Structure_Engine: Add method to check loads for id assignment
Add method to check loads for id assignment to prevent Exception raised when loads are defined before having done an initial push of the geometry.

For example, when exporting bar with loads, if you try  to export both bar and load simultaneously with the same Push component you will get a `the given key is not present in the Dictionary` error on it. This is not informative to newcomers. Potential actions:

1. The exception should be caught or, better, prevented. A method should be checking loads to see if they are to be applied to bar which do not yet have an Id assigned (`bar.CustomData[AdapterId]` not existing).

2. A more meaningful error message should appear on the Push component, explaining that a first Push is needed in order to assign Ids to the bars, and then a second push component may be used to export Loads.

### Test file

This is an example file with an ETABS export.

https://burohappold.sharepoint.com/:u:/s/BHoM/ES01T4axRFxMjHx_CJ0UzhEBNebevX0PjQ_sWKSp5Xm_Lw?e=XuIIbC

The exception mentioned above happens in the `SetLoad` method in the `Load.cs` file when `SetLoadDistributed` is called:
``` 
ret = model.FrameObj.SetLoadDistributed(bar.CustomData[AdapterId].ToString(), ...
```
as `CustomData` is not already defined.

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
### Issues addressed by this PR

Closes #875

Adding method checking if obejcts on the load all contain an specific adapter id.

### Test files

https://burohappold.sharepoint.com/:f:/s/BHoM/Ei47BMo0kzhNpmJDXFRIHlABU8sbHtLiiE-ELNQ12IwLXw?e=NAdF1b

### CODE REVIEW

**Comment by IsakNaslundBh** [comment]
@peterjamesnugent @JosefTaylor would any of you be able to have a quick look at this? Postscript exists. Should hopefully be a 5 min job.

File: Structure_Engine/Query/HasAssignedObjectIds.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+namespace BH.Engine.Structure
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Checks if the objects assigned to the IElementLoad all have an identifier matching the 'adapterIdName' assigned.\n" +
```
Comment (peterjamesnugent) [inline Structure_Engine/Query/HasAssignedObjectIds.cs:39]: AdapterIdName

File: Structure_Engine/Query/HasAssignedObjectIds.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Checks if the objects assigned to the IElementLoad all have an identifier matching the 'adapterIdName' assigned.\n" + 
+                     "Loads without correctly assigned ids to the objects can generally not be pushed through to structural packages.")]
+        [Input("load", "The load to check for id assignment.")]
+        [Input("adapterIdName", "The identifier key to check. For example 'Robot_id' for Autodesk Robot.")]
+        [Output("result", "Returns true if all objects assigned to the load has an id matching the 'adapterIdName' assigned.")]
```
Comment (peterjamesnugent) [inline Structure_Engine/Query/HasAssignedObjectIds.cs:43]: AdapterIdName - no need for '

**Review by peterjamesnugent — CHANGES_REQUESTED** [review]
Functionality works as expected, uploaded a new version of the script that works with the current UI.

Changes for description.

File: Structure_Engine/Query/HasAssignedObjectIds.cs (hunk 1 above)
Comment (IsakNaslundBh) [reply Structure_Engine/Query/HasAssignedObjectIds.cs]: was refering to the input name, but do get your point. Have updated

**Review by peterjamesnugent — APPROVED** [review]
LGTM.

**Pull request merged**